debezium-core/src/main/java/io/debezium/util/Threads.java

```diff
@@ -178,6 +181,81 @@ public static Thread timeout(String threadName,
         return new Thread(r, threadName);
     }
 
+    /**
+     * Return a new {@link ThreadFactory} that creates threads with names that begin with the given prefix and end with
+     * an monotonically increasing integer value.
+     * 
+     * @param threadNamePrefix the prefix for all thread names; may not be null
+     * @return the new thread factory; never null
+     */
+    public static ThreadFactory createFactory(String threadNamePrefix) {
+        AtomicInteger counter = new AtomicInteger();
```

Comment:
since `counter` is created on each method invocation, not sure I follow the logic behind the `monotonically increasing` statement